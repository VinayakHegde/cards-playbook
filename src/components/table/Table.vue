<style lang="scss" scoped src="./style.scss"></style>
<template>
  <ul v-if="$store.getters.cards.length" class="table">
    <li
      v-for="card in $store.getters.cards"
      :key="JSON.stringify(card)"
      class="card"
    >
      <div class="value" v-bind:class="card.name">{{ card.value }}</div>
      <div class="value bottom" v-bind:class="card.name">
        {{ card.value }}
      </div>
    </li>
  </ul>
</template>
