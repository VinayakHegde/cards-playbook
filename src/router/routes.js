export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
