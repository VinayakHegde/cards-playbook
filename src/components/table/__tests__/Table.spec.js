import Vuex from "vuex";
import VueRouter from "vue-router";
import { mount, createLocalVue } from "@vue/test-utils";
import storeConfig from "@/store/store-config";
import routes from "@/router/routes";
import Table from "../Table.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
describe("Table.vue", () => {
  it("match the snapshot", () => {
    const store = new Vuex.Store(storeConfig);

    const router = new VueRouter({ routes });
    const wrapper = mount(Table, {
      localVue,
      router,
      store
    });
    expect(wrapper).toMatchSnapshot();
  });
});
