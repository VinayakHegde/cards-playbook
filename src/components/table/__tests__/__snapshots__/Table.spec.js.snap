// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.vue match the snapshot 1`] = ``;
